### ISSUE TITLE
Doesn't handle includes with file systems properly

### ISSUE THREAD

**Issue by BinaryAura** [Enhex/premake-cmake#2]
When trying to use this module to make a CMakeLists.txt workspace for CLion. CMake errors when searching for a relative file in a subdirectory saying it cannot find the source file. It appears that CMake isn't properly desending into the subdirectory like it would in not cmake modes. My project does build when calling premake5 gmake.

Project
  premake5.lua
  src
    main.cpp
  deps
    premake5.lua
    a_library
      premake5.lua <- error occurs here with project > files

**Comment by Enhex** [comment]
nothing comes to mind as for what could've gone wrong.
perhaps you could create a minimal reproducible example of the problem and share it?

**Comment by BinaryAura** [comment]
The structure isn't minimal but the premake5.lua project is:

[HadronTest.zip](https://github.com/Enhex/premake-cmake/files/6411772/HadronTest.zip)

What I've noticed is that the subdirectory project in deps/hadron_engine is 'include' instead of 'add_subdirectory' in cmake meaning that cmake isn't adding to the relative paths. In premake files are translated directly over disregarding the orginal file structure. I believe cmake_workspace.lua:101 is the culprit. Anything that is directory that is included in premake should be add_subdirectory in cmake but, just changes cmake_workspace.lua:101 results in cmake attempting to add Test.cmake as a directory instead of including it.

**Comment by BinaryAura** [comment]
I messed with the output cmake a little to get it to work and I did the following:

HadronTest/CMakeLists.txt:28
include(deps/hadron_engine/Hadron.cmake)
add_subdirectory(deps/hadron_engine)

Add HadronTest/deps/hadron_engine/CMakeLists.txt
include(Hadron.cmake)

This built and compiled.
Now to figure out how to implement this in the script. I'll be using a fork and submitted a pull request. Which I'll link to.
